Add unified LoadingOverlay and integrate loading animations

HB616 opened the pull request:
Assisted by AI

**What has been changed?**
This PR introduces a shared and unified loading system across the application by adding reusable loading composables and integrating them into screens that previously lacked consistent loading feedback.

The goal is to ensure a consistent user experience, avoid duplicated loading implementations, and clearly communicate background operations (data loading, saving, refreshing) to the user.

**Key updates**
1. New shared loading components
• Added a LoadingOverlay composable to display a full-screen, blocking loading state with a dimmed background

2. Integration into existing screens
• EditEvent
• Calendar
• EditOverView
• ReplacementEmployee

**Why was this change made?**
Previously:
• Loading indicators were inconsistent across screens
• Some screens had no visual feedback during network or repository operations
• Different loading styles risked confusing users
• Adding loading to new screens required custom, duplicated implementations

This refactoring:
• Enforces a single loading design across the app
• Improves UX by clearly indicating background operations
• Makes loading behavior predictable and easier to maintain
• Aligns with MVVM best practices by driving UI from ViewModel state
• Prepares the app for future async-heavy features

**demo:**
<img width="393" height="436" alt="Capture d’écran 2025-12-16 à 18 23 49" src="https://github.com/user-attachments/assets/ed51a381-38a1-4993-b4a4-5d6db1df2dd9" />
<img width="344" height="721" alt="Capture d’écran 2025-12-17 à 21 30 33" src="https://github.com/user-attachments/assets/5d765326-9983-41b4-af8f-ed633a9e60ab" />

**Checklist**
- [ ] Added shared Loading composable
- [ ] Added shared LoadingOverlay composable
- [ ] Integrated loading into Edit Event flow
- [ ] Integrated loading into Calendar screen
- [ ] Ensured consistent behavior across sub-screens

**Relations**
• Task: Unify loading UI across screens - #366 
• User story: As a user, I want clear feedback when the app is loading so I understand what is happening and do not interact with incomplete screens.

HHHb616 committed:
add: Add unified LoadingOverlay and integrate loading animations across screens

     - Introduced a reusable Loading and LoadingOverlay composable
     - Display loading overlay when ViewModel is in loading state
     - Added missing loading indicators to Calendar, EditEvent and related flows
     - Ensured consistent loading UX across main and sub-screens

noa-flo reviewed (changes requested):
This PR introduces a good idea and is well done, but it misses explicit test for it.
Especially, I am concern on how it looks in context over a "real" screen. If the current tests already test it, add a screenshot of such a scenario, if not, add relevent tests.

HHHb616 committed:
fix: fix the issues mentioned by review

- add some test for appliesFilters

HB616 commented:
> This PR introduces a good idea and is well done, but it misses explicit test for it. Especially, I am concern on how it looks in context over a "real" screen. If the current tests already test it, add a screenshot of such a scenario, if not, add relevent tests.

I added a loading test — you can check the demo for a rough idea (I added a new image). The background slightly darkens and a loading indicator appears, but it disappears almost instantly; it’s more like a very quick transition animation.

HB616 commented:
> > This PR introduces a good idea and is well done, but it misses explicit test for it. Especially, I am concern on how it looks in context over a "real" screen. If the current tests already test it, add a screenshot of such a scenario, if not, add relevent tests.
> 
> I added a loading test — you can check the demo for a rough idea (I added a new image). The background slightly darkens and a loading indicator appears, but it disappears almost instantly; it’s more like a very quick transition animation.

<img width="344" height="721" alt="Capture d’écran 2025-12-17 à 21 30 33" src="https://github.com/user-attachments/assets/5d765326-9983-41b4-af8f-ed633a9e60ab" />

noa-flo reviewed (changes requested):
All my questions where answered, now the CI need to pass so we can check the SonarCloud status.

HB616 committed:
Merge branch 'main' into feature/unified-loading

HHHb616 committed:
fix: fix the issues mentioned by review

- format problem

Qoookie committed:
Merge branch 'main' into feature/unified-loading

Commit:
fix: refactore code

Qoookie commented:
I cannot merge it, I need Noa approval, for me every thing is ok, if you wake up before 8:45 you can merge otherwise, not a big deal !

noa-flo reviewed (approved):
The CI pass, no SonarCloud issues and the line coverage is above 80%, I aprove.
